src/snark/libsnark/common/utils.hpp

```diff
@@ -20,14 +20,14 @@ namespace libsnark {
 
 typedef std::vector<bool> bit_vector;
 
-/// returns ceil(log2(n)), so 1ul<<log2(n) is the smallest power of 2, that is not less than n
+/// returns ceil(log2(n)), so UINT64_C(1)<<log2(n) is the smallest power of 2, that is not less than n
 size_t log2(size_t n);
```

Comment:
Should these be ``uint64_t`` rather than ``size_t``? Ditto for ``exp2`` and for the ``n`` argument and return type of ``bitreverse``.